scipy/sparse/_coo.py

```diff
@@ -286,13 +293,17 @@ def toarray(self, order=None, out=None):
         fortran = int(B.flags.f_contiguous)
         if not fortran and not B.flags.c_contiguous:
             raise ValueError("Output array must be C or F contiguous")
-        if self.ndim > 2:
-            raise ValueError("Cannot densify higher-rank sparse array")
         # This handles both 0D and 1D cases correctly regardless of the
         # original shape.
-        M, N = self._shape_as_2d
-        coo_todense(M, N, self.nnz, self.row, self.col, self.data,
-                    B.ravel('A'), fortran)
+        if self.ndim < 3:
```

Comment:
Let's call `coo_todense_nd` for the 1-dimensional case as well, because otherwise we have to allocate a temporary array of all zeros when we access `self.row`.

Reply:
Addressed at 69e8757.